parsers/CY.py

```diff
@@ -63,17 +63,16 @@ def parse_production(self, html, capacity: dict) -> list:
                 elif col == 'Συμβατική Παραγωγή':
                     production['oil'] = float(val)
                 elif col == 'Εκτίμηση Διεσπαρμένης Παραγωγής (Φωτοβολταϊκά και Βιομάζα)':
-                    # Because solar is explicitly listed as "Solar PV" (so no thermal with energy storage) and there
-                    # is zero sunlight in the middle of the night (https://www.timeanddate.com/sun/cyprus/nicosia),
-                    # we use the biomass+solar generation reported at 0:00 to determine the portion of biomass+solar
-                    # which constitutes biomass
-                    if len(data) == 0:
-                        midnight_biomass = float(val)
-                        production['biomass'] = midnight_biomass
-                        production['solar'] = 0.0
-                    else:
-                        production['biomass'] = midnight_biomass
-                        production['solar'] = float(val) - midnight_biomass
+                    # Because solar is explicitly listed as "Solar PV" (so no thermal with energy storage)
+                    # and there is no sunlight between 10pm and 3am (https://www.timeanddate.com/sun/cyprus/nicosia),
+                    # we use the nightly biomass+solar generation reported to determine the portion of biomass+solar
+                    # which constitutes biomass.
+                    value = float(val)
+                    hour = datum['datetime'].hour
+                    if hour < 3 or hour >= 22:
```

Comment:
Do you know how the source handles daylight saving time? :)

Reply:
The source has table cells like `2021-05-26 00:00:00` which I suppose include DST. We set the timezone to Asia/Nicosia so I suppose `arrow` will take care of it.